features/upgrade/storage/upgrade.feature

```diff
@@ -172,3 +172,141 @@ Feature: Storage upgrade tests
+    Given the master version >= "4.5"
+ 
+    #Deploy csi hostpath driver 
+    When I run the :new_project client command with:
+      | project_name | csihostpath |
+    
+    Given I switch to cluster admin pseudo user		
+    When I use the "csihostpath" project
+
+    Given I obtain test data file "storage/csi/csi-rbac.yaml"   
+    When I run the :apply client command with:
+      | f | csi-rbac.yaml | 
+    Then the step should succeed
+    Given SCC "privileged" is added to the "csi-provisioner" service account
+    Given SCC "privileged" is added to the "csi-attacher" service account
+    Given SCC "privileged" is added to the "csi-snapshotter" service account
+    Given SCC "privileged" is added to the "csi-plugin" service account
+
+    Given I obtain test data file "storage/csi/csi-hostpath-attacher.yaml"
+    When I run the :apply client command with:
+      | f | csi-hostpath-attacher.yaml |
+    Then the step should succeed
+
+    Given I obtain test data file "storage/csi/csi-hostpath-driverinfo.yaml"
+    When I run the :apply client command with:
+      | f | csi-hostpath-driverinfo.yaml |
+    Then the step should succeed
+
+    Given I obtain test data file "storage/csi/csi-hostpath-plugin.yaml"
+    When I run the :apply client command with:
+      | f | csi-hostpath-plugin.yaml |
+    Then the step should succeed
+
+    Given I obtain test data file "storage/csi/csi-hostpath-provisioner.yaml"
+    When I run the :apply client command with:
+      | f | csi-hostpath-provisioner.yaml |
+    Then the step should succeed
+
+    Given I obtain test data file "storage/csi/csi-hostpath-snapshotter.yaml"
+    When I run the :apply client command with:
+      | f | csi-hostpath-snapshotter.yaml |
+    Then the step should succeed
+
+    And I wait up to 360 seconds for the steps to pass:
+    """
+    Given the pod named "csi-hostpath-attacher-0" is ready
+    Given the pod named "csi-hostpath-provisioner-0" is ready
+    Given the pod named "csi-hostpath-snapshotter-0" is ready
+    Given the pod named "csi-hostpathplugin-0" is ready
+    """
+    
+    #Create storageclass volumesnapshotclass
+    Given I obtain test data file "storage/csi/csi-storageclass.yaml"
+    When I run the :apply client command with:
+      | f | csi-storageclass.yaml |
+    Then the step should succeed
+
+    Given I obtain test data file "storage/csi/snapshotclass.yaml"
+    When I run the :apply client command with:
+      | f | snapshotclass.yaml |
+    Then the step should succeed
+
+    #Create pvc/pod/volumesnapshot
+    Given I obtain test data file "storage/misc/pvc-with-storageClassName.json"
+    When I run oc create over "pvc-with-storageClassName.json" replacing paths:
+      | ["metadata"]["name"]         | pvc-hostpath    |
+      | ["spec"]["storageClassName"] | csi-hostpath-sc |
+
+    Given I obtain test data file "storage/misc/pod.yaml"
```

Comment:
suggest using deployment instead of pod.

Reply:
I realize it is only for a pvc snapshot test (to write something), in this case, I think pod is ok for me.